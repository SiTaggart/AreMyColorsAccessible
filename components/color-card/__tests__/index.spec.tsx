/* eslint-env jest */

import React from 'react';
import ReactDOM from 'react-dom';
import ColorCard, { ColorCardProps } from '..';
import renderer from 'react-test-renderer';

describe('ColorCard', (): void => {
  const mockProps: ColorCardProps = {
    accessibility: {
      aa: true,
      aaLarge: true,
      aaa: true,
      aaaLarge: true
    },
    background: '#fff',
    color: '#000',
    contrast: 21
  };

  it('renders without crashing', (): void => {
    ReactDOM.render(<ColorCard {...mockProps} />, document.createElement('div'));
    const colorCard = renderer.create(<ColorCard {...mockProps} />).toJSON();
    expect(colorCard).toMatchSnapshot();
  });

  it('should render a not important card', (): void => {
    const colorCard = renderer.create(<ColorCard {...mockProps} isNotImportant={true} />).toJSON();
    expect(colorCard).toMatchSnapshot();
  });
});
