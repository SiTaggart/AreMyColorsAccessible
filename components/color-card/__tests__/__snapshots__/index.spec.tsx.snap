// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard renders without crashing 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-height: 8rem;
  min-width: 12rem;
  overflow: hidden;
  text-align: center;
  -webkit-transition: opacity ease-in;
  transition: opacity ease-in;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-4 {
  background-color: #F7F8F8;
  color: #666D70;
  border-radius: 3px;
  display: inline-block;
  font-size: 12px;
  padding: 1px 4px;
}

.emotion-8 {
  color: #000;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-weight: bold;
  font-size: 2rem;
  text-align: center;
}

.emotion-12 {
  background-color: #CFFCDA;
  color: #004215;
  border-radius: 3px;
  display: inline-block;
  font-size: 12px;
  padding: 1px 4px;
}

<div
  className="emotion-0 emotion-1"
  data-testid="colorCard"
>
  <div
    className="emotion-2 emotion-3"
  >
    <span
      className="emotion-4 emotion-5"
      title="Color contrast ratio"
    >
      21
       : 1
    </span>
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <div
      className="emotion-8 emotion-9"
      color="#000"
      data-testid="colorCard-swatch"
    >
      Yup
    </div>
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <span
      className="emotion-12 emotion-5"
    >
      Small: 
      AAA
    </span>
    <span
      className="emotion-12 emotion-5"
    >
      Large: 
      AAA
    </span>
  </div>
</div>
`;

exports[`ColorCard should render a not important card 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-height: 8rem;
  min-width: 12rem;
  overflow: hidden;
  opacity: 0.1;
  text-align: center;
  -webkit-transition: opacity ease-in;
  transition: opacity ease-in;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-4 {
  background-color: #F7F8F8;
  color: #666D70;
  border-radius: 3px;
  display: inline-block;
  font-size: 12px;
  padding: 1px 4px;
}

.emotion-8 {
  color: #000;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-weight: bold;
  font-size: 2rem;
  text-align: center;
}

.emotion-12 {
  background-color: #CFFCDA;
  color: #004215;
  border-radius: 3px;
  display: inline-block;
  font-size: 12px;
  padding: 1px 4px;
}

<div
  className="emotion-0 emotion-1"
  data-testid="colorCard"
>
  <div
    className="emotion-2 emotion-3"
  >
    <span
      className="emotion-4 emotion-5"
      title="Color contrast ratio"
    >
      21
       : 1
    </span>
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <div
      className="emotion-8 emotion-9"
      color="#000"
      data-testid="colorCard-swatch"
    >
      Yup
    </div>
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <span
      className="emotion-12 emotion-5"
    >
      Small: 
      AAA
    </span>
    <span
      className="emotion-12 emotion-5"
    >
      Large: 
      AAA
    </span>
  </div>
</div>
`;
